include/ttmlir/Dialect/TTNN/IR/TTNNOps.td

```diff
@@ -27,13 +27,36 @@ def TTNN_GetDeviceOp : TTNN_Op<"get_device"> {
     let results = (outs TT_Device:$device);
 }
 
+def TTNN_CompositeToLayoutOp : TTNN_Op<"composite_to_layout"> {
```

Comment:
Is this going to be permanent?  Perhaps we can replace this with ToLayout? As I understand it TTNN folks plan to make `ttnn::to_layout` flexible enough to support all kinds of layout transitions.  In the meantime we can specialize ToLayout to the respective lowered op.

Reply:
@nsmithtt that's a good point. I guess I can rename this op to ToLayout, and then the downstream decomposed ToLayout to something like TTNN_TilizeUntilize. Later when `ttnn::to_layout` supports all combinations of parameters, we can simply remove the downstream decomposition pass and use this op instead.

Reply:
Yeah if you are going to change the name it's good idea to change downstream decomposed to something else.

Reply:
Or we just leave it?  At the initial conversion / optimizer just targets:
- `ToLayout`

Then, in the short term, we might have to lower `ToLayout` to one of:
- `ToMemoryConfig`
- `Typecast`
- etc.
- Or just leave it, if it's supported, as `ToLayout`

Over time, we can just erase the specialized lowering for cases where `ToLayout` has added support for it.

Reply:
I don't have a hard preference here but leaving the same name is in my opinion bug prone. Reordering passes may create hard to debug issues with Op which has different meaning depending on place it's used.

Reply:
@jnie-TT yeah that's right.

@nobradovictt, I agree it's not ideal, but neither is introducing a dummy op esp if the eventual semantics of the `ToLayout` op are exactly how we want to use it.  Think of it as a workaround, we lower to `ToLayout`, workaround passes at the end get applied that specialize out all the cases that `ToLayout` doesn't support yet.

Reply:
@svuckovicTT I could make the same argument against having an op of the same name as in TTNN lib which carries completely different meaning(that it pushes TTNN dialect further from TTNN lib). I personally actually find it even worse, it's "wrong result" but insidious. It's easy to assert that dummy op is not expected in final IR.
But as this is a temp workaround, and more of you seems to prefer same name option, Im fine with that.

Reply:
I'll create a separate issue to track this.